services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -379,9 +425,14 @@ async def delete_file(self, user_id: UserID, file_id: StorageFileID):
                 file: FileMetaDataAtDB = await db_file_meta_data.get(
                     conn, parse_obj_as(SimcoreS3FileID, file_id)
                 )
-                await get_s3_client(self.app).delete_file(
-                    file.bucket_name, file.file_id
-                )
+                if file.is_directory:
+                    await get_s3_client(self.app).delete_files_in_path(
```

Comment:
so here I see why you duplicated the function.
I would propose to have only one function:
- that uses pagination so you are sure to delete everything inside
- that gets a parameter such as ```recursive:bool=False```
- if recursive is True, then you delete everything inside
- the function makes itself use of ```ensure_ends_with```.